Add merged json_encoders inheritance

art049 opened the pull request:
## Change Summary

While inheriting the configuration , the `json_encoders` dictionaries are merged and the child classes can still override an encoder  defined in the parent class.

## Related issue number

Fixes #2024 

## Checklist

* [x] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable
* [x] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)

samuelcolvin commented on pydantic/main.py:
```diff
@@ -158,13 +158,20 @@ def prepare_field(cls, field: 'ModelField') -> None:
     else:
         base_classes = self_config, parent_config  # type: ignore
-    return type('Config', base_classes, {})
+        json_encoders = {
+            **getattr(parent_config, 'json_encoders', {}),
+            **getattr(self_config, 'json_encoders', {}),
+        }
+        namespace['json_encoders'] = json_encoders
```
```suggestion
        namespace['json_encoders'] = {
            **getattr(parent_config, 'json_encoders', {}),
            **getattr(self_config, 'json_encoders', {}),
        }
```

samuelcolvin reviewed (commented):
you need to add a change description and update the docs to describe the new behaviour preferably with an example

art049 commented:
> you need to add a change description and update the docs to describe the new behaviour preferably with an example
Thanks for the feedback @samuelcolvin  !
I wasn't very sure this feature was ok as I didn't have any feedback neither on the issue nor the PR so I didn't complete it but as it seems ok to you I will be happy complete the changes 😄

samuelcolvin commented:
@art049, I'm happy to accept this change.

We still need a change description and docs.

samuelcolvin commented:
@art049 are you okay to fix this?

art049 commented:
> @art049 are you okay to fix this?
@samuelcolvin yes, totally.
I fixed the code smell and added the required documentation :smile: .

WDYT about the docs? I don't know if it explains the feature properly enough

samuelcolvin commented:
great, thank you so much.
